cdap-watchdog/src/test/java/co/cask/cdap/logging/plugins/RollingLocationLogAppenderTest.java

```diff
@@ -213,6 +214,38 @@ public void testRollOver() throws Exception {
     Assert.assertEquals(10, parentDir.list().size());
   }
 
+  @Test
+  public void testFileClose() throws Exception {
+    // assume SLF4J is bound to logback in the current environment
+    AppenderContext appenderContext = new LocalAppenderContext(injector.getInstance(DatasetFramework.class),
+                                                               injector.getInstance(TransactionSystemClient.class),
+                                                               injector.getInstance(LocationFactory.class),
+                                                               new NoOpMetricsCollectionService());
+
+    JoranConfigurator configurator = new JoranConfigurator();
+    configurator.setContext(appenderContext);
+    // Call context.reset() to clear any previous configuration, e.g. default
+    // configuration. For multi-step configuration, omit calling context.reset().
+    appenderContext.reset();
+
+    configurator.doConfigure(getClass().getResourceAsStream("/rolling-appender-logback-test.xml"));
+    StatusPrinter.printInCaseOfErrorsOrWarnings(appenderContext);
+
+    RollingLocationLogAppender rollingAppender =
+      (RollingLocationLogAppender) appenderContext.getLogger(RollingLocationLogAppenderTest.class)
+        .getAppender("rollingAppender");
+
+    addTagsToMdc("testNs", "testApp");
+    Logger logger = appenderContext.getLogger(RollingLocationLogAppenderTest.class);
+    ingestLogs(logger, 20);
+
+    // wait for 1 min so that file is eligible for closing
+    Thread.sleep(TimeUnit.MINUTES.toMillis(1));
```

Comment:
set the file inactive time to be 200 milli seconds instead of 20 seconds in the test. sleep for like 500 milliseconds and then perform the check.